sqlglot/transforms.py

```diff
@@ -81,3 +81,15 @@ def _transform(self, *args, **kwargs):
 
 
 UNALIAS_GROUP = {exp.Group: preprocess([unalias_group], delegate("group_sql"))}
+ADD_TO_DPIPE = {
+    exp.Add: lambda self, e: self.dpipe_sql(e)
+    if e.type
+    in [
+        exp.DataType.Type.VARCHAR,
+        exp.DataType.Type.NVARCHAR,
+        exp.DataType.Type.CHAR,
+        exp.DataType.Type.NCHAR,
+        exp.DataType.Type.TEXT,
+    ]
```

Comment:
Maybe we could export some helper functions to help us identify an expression's general type class (e.g if it's a string, a numeric, date etc), assuming we've already inferred its type. I think that might be useful for other type inference applications we might have in the future.

Otherwise, If that's not something we should prioritize right now, you could convert that list into a set.

Reply:
Indeed, would be nice to have such a supertype available. For now i converted it into a set. Also i rewrote the approach a bit to also be able to use the function within test_transforms.py.

Reply:
> For now i converted it into a set.

I think you wrote it as a tuple instead. Was this a typo?